CONTRIBUTING.md

```diff
@@ -0,0 +1,67 @@
+[![Version](https://img.shields.io/npm/v/@hotwired/turbo)](https://www.npmjs.com/package/@hotwired/turbo)
+[![License](https://img.shields.io/github/license/hotwired/turbo)](https://github.com/hotwired/turbo)
+
+# Contributing
+
+Please note we have a [code of conduct](https://github.com/hotwired/turbo/blob/main/CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
+
+## Sending a Pull Request
+The core team is monitoring for pull requests. We will review your pull request and either merge it, request changes to it, or close it with an explanation.
+
+Before submitting a pull request, please make sure the following is done:
+
+1. Fork the repository and create your branch from main.
+2. Run `yarn` in the repository root.
+3. If you’ve fixed a bug or added code that should be tested, add tests!
+4. Ensure the test suite passes (`yarn build && yarn test`).
+
+## Developing locally
+
+During the process of developing the library locally we first have to check out the repository and create a branch from main.
+
+```bash
+git clone https://github.com/hotwired/turbo.git
+cd turbo
+yarn
+```
+
+```bash
+git checkout -b '<your_branch_name>'
+```
+
+Once you are done developing the feature or bug fix you have 2 options:
+1. Run the test suite
+2. Run a local webserver and checkout your changes manually
+
+### Testing
+The library is tested by running the test suite (found in: `src/tests/*`) against headless browsers. The browsers are setup in `intern.json` check it out to see the used browser environments.
+
+The tests are using the compiled version of the library and they are themselves also compiled. To compile the tests and library:
+
+```bash
+yarn build
+```
+
+> **_Caveat:_**  Don't forget to run the build command before you run the tests everytime you changed anything, if you forget it you will eventually run into unexpected results.
+
+To run the tests:
+
+```bash
+yarn test
+```
+
+### Test files
+Please add your tests in the test files closely related to the feature itself. For example when touching the `src/core/drive/page_renderer.ts` youre test will probably endup in the `src/tests/functional/rendering_tests.ts`.
+
+The html files needed for the tests are stored in: `src/tests/fixtures/`
+
+### Local webserver
+Since the tests are running in headless browsers it's not easy to debug them easily without using the debugger. Sometimes it's easier to run the supplied webserver and manually click through the test fixtures.
+
+To run the webserver:
+
+```bash
+yarn start
```

Comment:
This requires a build (via `yarn build`), or a separate process running `yarn watch`.

Reply:
Depending on their operating system, a developer might be able to execute `open http://localhost:9000/src/tests`, and then browse from there.

Reply:
Unfortunately the the webserver is not listing the directory contents for index less directories. When opening the page on: http://localhost:9000/src/tests they will see the following error message: `404 Not Found: /src/tests/`

I'm a bit unsure if this causes extra confusion or not, what do you think?